cms/models.py

```diff
@@ -835,6 +844,15 @@ def course_lineup(self):
         """Gets the course carousel page"""
         return self.program_page.course_lineup if self.program_page else None
 
+    @property
+    def news_and_events(self):
+        """
+        Gets the news and events section subpage
+        """
+        if self.program_page and self.program_page.news_and_events:
```

Comment:
I think the same logic would not work with external courses.

Reply:
Good catch, Yeah forgot to move corresponding override there.

Reply:
but for point#2, I need to ask from @briangrossman, It is bit tricky because we do not have any relationship in database for external courses with external program. I am not sure either we need to look over this edge case in case of external program and courses ? 

I did not assume external courses and external program by this statement below earlier either.
`if the course is a part of a program, it won't have it's own carousel but will use the one defined for the parent program`

Reply:
@asadiqbal08 
You are correct that there isn't a relationship between external courses and external programs in our database. We are overriding the contents as @arslanashraf7 explains above to create courseware carousels for programs.

If you are able to figure out a way to determine that an external course is a part of an external program using the course carousel, that would be ideal.

If it's not feasible to make that work, it is fine to not implement this functionality and we can build separate News and Events features for each external course and program.

Thanks!

Reply:
@briangrossman , I looked around but it can be tricky under this scenario if the external course is associated with more than 1 external programs then it can be difficult to choose the appropriate carousal. If that really required then we can open a separate story for discussing it.